src/vs/platform/window/common/window.ts

```diff
@@ -279,7 +283,8 @@ export interface INativeWindowConfiguration extends IWindowConfiguration, Native
 
 	profiles: {
 		all: readonly UriDto<IUserDataProfile>[];
-		current: UriDto<IUserDataProfile>;
+		profile?: UriDto<IUserDataProfile>;
+		profileOptions?: ProfileOptions;
```

Comment:
The distinction between `profile` and `profileOptions` is not clear to me.

Reply:
`profile` is the current associated profile and `profileOptions` are profile configuration options to use for the current workspace.

Reply:
This is still confusing to me and I think we should change semantics here. Because it looks to me as if I am passing in options for the profile of the window but in fact I am passing in the name of the profile to use. Imho this is now incredibly hard to tell apart whether `profile` is used or `profileOptions`.

Let us keep the semantic of `options` being something that configures `something` where `something` is `profile`. And if in fact this is for telling the window which profile to use by a name, then imho we should make `profile` a `IUserDataProfile | string` so that we know its either going to be an existing profile or a new profile.

Again, I think if the main process was to ensure the profile is there, then this would be much nicer, because instead of passing in the name of a new profile to create, it would be just the profile.

Reply:
Changed it to following

```
profiles: {
		all: readonly UriDto<IUserDataProfile>[];
		workspace: UriDto<IUserDataProfile> | IUserDataProfileInfo;
	};
```

Named the property to `workspace` that contains the profile or profile info to be used for the workspace.